cpp/ql/src/Best Practices/Hiding/DeclarationHidesParameter.ql

```diff
@@ -11,6 +11,20 @@
 
 import cpp
 
+/**
+ * Gets the template that a function `f` is constructed from, or just `f` if it
+ * is not from a template instantiation.
+ */
+Function getConstructedFrom(Function f) {
```

Comment:
Added.

I wish we had similar explanations for the two uses of `getDefinitionLocation()`.  I'm suspicious they can both be removed now (following my change) in favour of a simple `strictcount(f.getDefinition()) = 1`, but it would require quite a bit of testing effort to build up confidence in this.